lib/identity/IdentityMerkDBRepository.js

```diff
@@ -0,0 +1,93 @@
+const MerkDbTransaction = require('../merkDb/MerkDbTransaction');
+
+class IdentityMerkDBRepository {
+  /**
+   *
+   * @param {Merk} identityMerkDB
+   * @param {DashPlatformProtocol} noStateDpp
+   */
+  constructor(identityMerkDB, noStateDpp) {
+    this.db = identityMerkDB;
+    this.dpp = noStateDpp;
+  }
+
+  /**
+   * Store identity into database
+   *
+   * @param {Identity} identity
+   * @param {MerkDbTransaction} [transaction]
+   * @return {Promise<IdentityMerkDBRepository>}
+   */
+  async store(identity, transaction = undefined) {
+    let db;
+
+    if (transaction) {
+      db = transaction.db;
+
+      db.put(
+        identity.getId(),
+        identity.toBuffer(),
+      );
+    } else {
+      db = this.db;
+
+      db
+        .batch()
+        .put(
+          identity.getId(),
+          identity.toBuffer(),
+        )
+        .commitSync();
+    }
+
+    return this;
+  }
+
+  /**
+   * Fetch identity by id from database
+   *
+   * @param {Identifier} id
+   * @param {MerkDbTransaction} [transaction]
+   * @return {Promise<null|Identity>}
+   */
+  async fetch(id, transaction = undefined) {
+    let db;
+    // const db = transaction ? transaction.db : this.db;
+
+    try {
+      let encodedIdentity;
+
+      if (transaction) {
+        db = transaction.db;
+
+        encodedIdentity = db.get(id);
```

Comment:
Why wy call methods in transaction DB differently? If we have the same namming you can just swap db instance.

Reply:
differently from what?

Reply:
From MerkDB interface.